tests/unittests/core/test_trial.py

```diff
@@ -208,6 +208,72 @@ def test_gradient_property(self, exp_config):
         assert t.gradient.value == [5, 3]
         tmp = exp_config[1][1]['results'].pop()
 
+    def test_constraints_property(self, exp_config):
+        """Tests the property for accessing constraints"""
+        # 0 result of type 'constraints' exist
+        tmp = exp_config[1][1]['results'].pop(1)
+        t = Trial(**exp_config[1][1])
+        assert t.constraints == []
+        exp_config[1][1]['results'].append(tmp)
+
+        # 1 result of type 'constraint' exist
+        expected = [
+            Trial.Result(name='contra', type='constraint', value=1.2)
+        ]
+
+        trial = Trial(**exp_config[1][1])
+
+        assert expected == trial.constraints
+
+        # > 1 results of type 'constraint' exist
+        expected = [
+            Trial.Result(name='contra', type='constraint', value=1.2),
+            Trial.Result(name='a', type='constraint', value=5),
+            Trial.Result(name='b', type='constraint', value=20)
+        ]
```

Comment:
You should add a result that is not a constraint to test that they are filtered out properly.

Reply:
Or maybe there are results already coming from exp_config? In this case I would be in favor of hard-coding the additional results here and using a blank trial if possible.

Reply:
Yeah, there is already results in this variable. It will be explicit when I replace _exp_config_ by trials.